### PR TITLE
Add missing networking support for Illumos/Solaris as used in quinn

### PR DESCRIPTION
**Pull request by psumbera**

### CODE REVIEW

**Comment by pfmooney** [comment]
`libc-test` still passes on illumos with this change in place

**Comment by psumbera** [comment]
@JohnTitor can you please have look at this? Thank you!

**Comment by psumbera** [comment]
@JohnTitor is there anything wrong with this pull request?

**Comment by psumbera** [comment]
r? @tgross35 

Can you please help with merging of this?

**Comment by tgross35** [comment]
I am not sure what to check against, are there docs/headers online for the constants? Changes look fine to me with one nit above.

Please rebase and squash so the merge commits go away.

File: src/unix/solarish/mod.rs [hunk 1]
```diff
@@ -2856,15 +2874,19 @@ extern "C" {
...
     pub fn bind(socket: ::c_int, address: *const ::sockaddr, address_len: ::socklen_t) -> ::c_int;
 
     pub fn writev(fd: ::c_int, iov: *const ::iovec, iovcnt: ::c_int) -> ::ssize_t;
     pub fn readv(fd: ::c_int, iov: *const ::iovec, iovcnt: ::c_int) -> ::ssize_t;
 
     #[cfg_attr(target_os = "illumos", link_name = "__xnet_sendmsg")]
+    #[cfg_attr(target_os = "solaris", link_name = "__xnet_sendmsg")]
     pub fn sendmsg(fd: ::c_int, msg: *const ::msghdr, flags: ::c_int) -> ::ssize_t;
     #[cfg_attr(target_os = "illumos", link_name = "__xnet_recvmsg")]
+    #[cfg_attr(target_os = "solaris", link_name = "__xnet_recvmsg")]
```
Comment (pfmooney) [inline src/unix/solarish/mod.rs:2887]: For cases where the `link_name` now applies to both illumos and Solaris, there should be no need for the `cfg_attr` test while in `unix/solarish/mod.rs`

**Comment by psumbera** [comment]
> I am not sure what to check against, are there docs/headers online for the constants?

The only way how to review header files I can think of is:
1. Login to Solaris 11.4 machines as they are available at https://portal.cfarm.net/machines/list/
2. Install Solaris CBE release per https://blogs.oracle.com/solaris/post/announcing-the-first-oracle-solaris-114-cbe

**Comment by tgross35** [comment]
I don't have a cfarm account so that would take a couple days. Since @pfmooney has already verified illumos, I have no objection to merging.

This should have an update `libc-test/semver`, otherwise LGTM.

**Review by tgross35 — APPROVED** [review]

**Pull request merged**